spec/lib/active_merchant/billing/affirm_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'spec_helper'
+
+RSpec.describe ActiveMerchant::Billing::Affirm do
+
+  let(:gateway_options) do
+    {
+      public_api_key: "XPSQ3CA7PLN7CJCK",
+      private_api_key: "w9mxkQUryKjTYDqOfSvYTeTGoLIURKpU",
+      test_mode: true
+    }
+  end
+
+  let(:affirm_payment_source) { create(:affirm_checkout, token: "MFLJS57G6BT5OQXN") }
+
+  subject do
+    ActiveMerchant::Billing::Affirm.new(gateway_options)
+  end
+
+  describe "#authorize" do
+    it "with correct payment source" do
+      VCR.use_cassette("valid_authorize") do
```

Comment:
I returns the order data I guess, I setup a payment in a sandbox to record the details since we can only use the javascript way of starting a payment in Affirm. Then I used that data to record the responses from Affirm.